sdk/eventhub/event-hubs/src/connectionContext.ts

```diff
@@ -48,6 +51,24 @@ export interface ConnectionContext extends ConnectionContextBase {
    * the underlying amqp connection for the EventHub Client.
    */
   managementSession?: ManagementClient;
+
+  /**
+   * Indicates whether the connection is in the process of closing.
+   */
+  isConnectionClosing(): boolean;
+  /**
+   * Resolves once the connectionContext is ready to be opened.
+   */
+  readyToOpen(): Promise<void>;
+  /**
+   * Resolves once the context's connection emits a `disconnected` event.
+   */
+  waitForConnectionDisconnected(): Promise<void>;
```

Comment:
`waitForDisconnectEvent` ? Just to ensure new contributors dont confuse this to mean that the `disconnected` function is done executing. Bonus points if we can rename `disconnected` to `onDisconnect`?

Reply:
I like it! Also updated `disconnected` to `onDisconnected` (note that I went with onDisconnected instead of onDisconnect because the event is actually `disconnected`.